### PR TITLE
Update and clarify GMM tutorial

### PR DESCRIPTION
**Pull request by fritzo**
Addresses #879 

- Updates the Gaussian Mixture Model tutorial to modern PyTorch
- Switches to parallel enumeration (twice as fast!)
- Renames variables to be clearer
- Clarifies an error message that would have made this process easier ;-)

### CODE REVIEW

**Comment by fritzo** [comment]
cc @vincent6606

File: tutorial/source/gmm.ipynb [hunk 1]
```diff
@@ -116,21 +122,22 @@
...
      "text": [
       "\n",
-      "...............................................................................\n",
-      "...............................................................................\n",
-      "......................................."
+      "...................................................................................................\n",
+      "...................................................................................................\n",
+      "...................................................................................................\n",
+      "...................................................................................................\n",
+      "..................................................................................................."
```
Comment (jpchen) [inline tutorial/source/gmm.ipynb]: ?

File: tutorial/source/gmm.ipynb [hunk 2]
```diff
@@ -144,15 +151,15 @@
...
...hRckSS+88IL+9V//NbL/xRdflDFG7733ntLT0yNdVhcyY4zmz5+vkSNH6rvf/W5kP/XYNceOHVN1dbUkqampSZs3b9bIkSN1zTXX6LXXXpN0aj221u9rr72mqVOnXvC9JEuXLtWhQ4dUXl6uNWvWaOrUqfr1r39NHXZBQ0OD6urqIp83bdqkMWPG9I2/5x4Z2beg9evXm4svvtjk5+ebJUuWxLo4fdbs2bNNVlaWcTqdJicnxzz//POmsrLSTJ061RQUFJgvfelLpqqqyhgTmuX8ne98x+Tn55sxY8aY999/P8al7xu2bt1qJJmxY8ea8ePHm/Hjx5v169dTj130l7/8xUyYMMGMHTvWjB492jzyyCPGGGP27dtnioqKzPDhw82sWbNMc3OzMcaYpqYmM2vWLDN8+HBTVFRk9u3bF8vi9zl//OMfI7PoqcPo7du3z4wbN86MGzfOjBo1KpIffeHvmaVqAQCIQ3TRAwAQhwh4AADiEAEPAEAcIuABAIhDBDwAAHGIgAcQMWXKFEmhJXRXrVoV49IAOB8EPICId999V9K5BXzrymcA+gYCHkBESkqKJGnx4sXaunWrJkyYoJ/97GcKBAJatGiRioqKNG7cOP385z+XJG3ZskVXXXWViouLNWrUKDU0NGjGjBkaP368xowZo1deeSWW/xzggsbLZgCc4oknntBPfvITvfHGG5Kk0tJSpaen6/3331dLS4u+8IUvaNq0aZKknTt36q9//asuuugi/fa3v1V2drbWr18vSaqpqYnZvwG40NGCB3BWmzZt0osvvqgJEyZo8uTJqqqq0t69eyVJkyZN0kUXXSRJGjt2rDZv3qzvf//72rp1q9LT02NZbOCCRsADOCtjjJ555hnt2rVLu3bt0v79+yMt+OTk5Mh5I0aM0M6dOzV27Fg9+OCDevTRR2NVZOCCR8ADOEVqamrkDVmSNH36dC1fvlw+n0+S9Pe//10NDQ2nfO/IkSNKSkrSN77xDS1atEg7d+7stTIDaI8xeACnGDdunBwOh8aPH6+5c+dq4cKFKi8v18SJE2WM0aBBg7R27dpTvve///u/WrRokex2u1wul5YvXx6D0gOQJN4mBwBAHKKLHgCAOETAAwAQhwh4AADiEAEPAEAcIuABAIhDBDwAAHGIgAcAIA79f6Gk+m1wMWKQAAAAAElFTkSuQmCC\n",
```
Comment (jpchen) [inline tutorial/source/gmm.ipynb]: would be nice to save to file and just embed the image here, like in vae tutorials

File: tutorial/source/gmm.ipynb (hunk 1 above)
Comment (fritzo) [reply tutorial/source/gmm.ipynb]: changed line length from 80 to 100; fixed typo; changed iters from 200 to 500.

File: tutorial/source/gmm.ipynb (hunk 2 above)
Comment (fritzo) [reply tutorial/source/gmm.ipynb]: What's the advantage of that? Note that I cannot see the VAE tutorial images locally when I use jupyter (maybe due to path errors).

**Pull request merged**